postgres-operator-sdk/README.md

```diff
@@ -0,0 +1,288 @@
+* Begin experiment.
+
+### Setup AWS EC2 Instance
+Go to AWS and create an EC2 instance with these specifications:
+- Ubuntu 18.04 image
+- m5d.xlarge instance
+- A minimum requirement of 16GB of RAM is required. The m5d.xlarge satisfies this requirement and offers around 150GB
+of SSD storage which should suffice. 
+
+### Setup Kubernetes on EC2 Instance and make sure it works
+Follow the article referred to above for detailed steps in initializing Kubernetes on your AWS instance.
+[Link Here](https://dzone.com/articles/easy-step-by-step-local-kubernetes-source-code-cha)
+
+### Modify Kubernetes to allow for API call detection
+Turn off Kubernetes instance if currently running.
+Navigate to kubernetes/staging/src/k8s.io/apiserver/pkg/server/filters/
+Open the wrap.go file.
+Find this line:
+```
+logger := httplog.NewLogged(req, &w)
+```
+Right before the above line, insert this code: 
+```
+urlPath := req.URL.String()
+
+if strings.Contains(urlPath, "postgres") {
+      glog.Infof("==========================================================")
+      glog.Infof("Hello hello Postgreses")                
+      glog.Infof("==========================================================")
+}
+```
+
+Note: Do not forget to import the strings library into wrap.go.
+
+## Clone Postgres Operator repo
+After setting up Kubernetes, we need to setup our repo. The guide in the article also deals with setting up the $GOPATH
+so a lot of the work is done thankfully. Clone the repo using: 
+
+```
+git clone https://github.com/NoahElzein/postgrescontroller
+```
+
+We need to install two dependencies for this repo. 
+
+[lib/pq](https://github.com/lib/pq)
+
+Clone the above repo under the github.com folder.
+
+```
+git clone https://github.com/lib/pq.git
+```
+
+OPTIONAL: You can install the Postgres command line tool to verify that Postgres container is running on Kubernetes:
+
+```
+sudo apt-get install postgresql postgresql-contrib
+```
+
+## Begin Experiment
+
+--> Say that you will need 4 windows - create them.
+
+Run the Kubernetes cluster as referred to in the article about setting up Kubernetes. 
+
+Run the Postgres Operator in a separate window as described earlier in the post. 
+
+Access the tmp folder and tail the apiserver logs in a third terminal window.
+
+```
+tail -f kube-apiserver.log
+```
+
+Open a fourth terminal terminal window
+
+Go into the Postgres Operator project
+
+Register a Postgres object using the kubectl command:
+
+```
+kubectl create -f deploy/crd.yaml
+```
+
+You should see the message we inserted in wrap.go pop up in the log file (The log terminal window). 
+
+Now it is time to begin the experiment:
+
+The way we will measure performance of the library is based on the number of API calls that appear over a period of time during experimentation. The reason for this is because at the moment there is no way to differentiate between Resync API calls and API calls resulting from Library calls in Postgrescontroller.
```

Comment:
The way ... experimentation. The reason we do this is because this allows the controller to reach a stable state after a custom resource is created/updated.
There are two types of API calls that the controller makes to the Kubernetes API server. Calls made in response to the action taken by the user and Resync calls that are made by the controller to periodically sync up its queue. There is no good way to distinguish between these calls on the Server side. So by allowing the time for the controller to settle, we measure the cumulative effect of user-triggered calls and the resync calls.